libyul/YulString.h

```diff
@@ -44,6 +44,12 @@ class YulStringRepository: boost::noncopyable
 	};
 
 	YulStringRepository() = default;
+	YulStringRepository& operator=(YulStringRepository&& _rhs)
```

Comment:
```suggestion
	YulStringRepository& operator=(YulStringRepository&& _rhs) = default;
```

Reply:
It's a little weird because it should not be deleted.

Reply:
If I default the move assignment operator, the compiler warns me that the explicit default will be implicitly deleted because the parent class (boost::noncopyable) explicitly deletes its own move assignment operator.

Reply:
Ah, I see! But then it is probably better to just implement a manual reset function.

Reply:
Really? Why does ``noncopyable`` explicitly delete the move assignment operator? That's a bit weird and contrary to the boost docs...

Reply:
@ekpyron @chriseth What do you mean by a manual reset function?

Reply:
The weird way to do it would be this :-D:
``void reset() { this->~YulStringRepository(); new (this) YulStringRepository(); }``
But that's actually a bit dangerous wrt exceptions, so we probably shouldn't do it.

But what you can do is just reset all members to their default-initialized state:
```
void reset() {
m_strings = std::vector<std::shared_ptr<std::string>> {std::make_shared<std::string>()};
m_hashToID = std::unordered_multimap<std::uint64_t, size_t> {{emptyHash(), 0}};
}
```

Reply:
Thank you @ekpyron . Will check if the problem reoccurs with such a manual reset.